### PR TITLE
[RFR] CM-249/ Store structure(session)

### PR DESCRIPTION
**Pull request by e-molochko**
This change is [Reviewable](https://reviewable.io/reviews/ozalexo/chronomint-rn/26)

### CODE REVIEW

File: src/redux/session/actions.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  return {
+    type: USER_ACTIONS.LOGIN,
+    currentWallet,
+  }
+}
+
+export const logout = () => {
+  return {
+    type: USER_ACTIONS.LOGOUT,
+    currentWallet: '',
```
Comment (ozalexo) [inline src/redux/session/actions.js]: I suggest to use null here. Empty string also falsy, but in this case we not handle error if we will have empty address for a reason.

File: src/redux/session/constants.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+export const DUCK_SESSION = 'session'
+
+export const USER_ACTIONS = {
```
Comment (ozalexo) [inline src/redux/session/constants.js]: Let's define casual naming standard for action constants.

For DUCK session we need to use something like this:
```
export const SESSION_LOGIN = 'SESSION/LOGIN'
export const SESSION_LOGOUT = 'SESSION/LOGOUT'
```

So, it should has the following format: export const DUCKPREFIX_ACTION_TITLE = 'DUCKPREFIX/ACTION_TITLE'

File: src/redux/session/initialState.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+export const initialState = {
+  session: {
+    currentWallet: '',
```
Comment (ozalexo) [inline src/redux/session/initialState.js]: See my previous comment about null and ''.

File: src/redux/session/selectors.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+import { createSelector } from 'reselect'
+import { DUCK_SESSION } from './constants'
+
+
+export const getCurrentWallet = createSelector(
+  (state) => state[DUCK_SESSION],
```
Comment (ozalexo) [inline src/redux/session/selectors.js]: I suggest to make selector for each duck. 
```
import { DUCK_SESSION } from './constants'

const getDuckSession = (state) =>
  state[DUCK_SESSION]

export const getCurrentWallet = createSelector(
  getDuckSession,
  (session) => {
    const { currentWallet } = session
    return currentWallet
  },
)
```

In very rare case we may want to have whole duck in JSX (in other ducks). in this case
```
const getDuckSession = (state) =>
  state[DUCK_SESSION]
```
became
```
export const getDuckSession = (state) =>
  state[DUCK_SESSION]
```

File: src/redux/session/thunks.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import { login, logout } from './actions'
+
+export const Login = (currentWallet) => async (dispatch) => {
```
Comment (ozalexo) [inline src/redux/session/thunks.js]: Let's add async when we really will have await.

File: src/redux/session/constants.js (hunk 2 above)
Comment (e-molochko) [reply src/redux/session/constants.js]: So mb it better to have an object of constants so our import will look as one row?
Like :
SESSION = {
LOGIN: 'LOGIN',
LOGOUT: 'LOGOUT',
}

File: src/screens/Login/Start/StartContainer.js [hunk 6]
```diff
@@ -28,6 +32,11 @@ class StartContainer extends PureComponent {
     }).isRequired,
   }
 
+  componentDidMount () {
+    store.dispatch(loginThunk('0xf1106d1eb597ef2f14c8f5343c1b4203fa0f2e9b')) //check login
```
Comment (ozalexo) [inline src/screens/Login/Start/StartContainer.js]: As far as I can see, these changes at lines 36-37 only for testing purposes...

File: src/screens/Login/Start/StartContainer.js [hunk 7]
```diff
@@ -6,8 +6,12 @@
 import React, { PureComponent } from 'react'
 import { connect } from 'react-redux'
 import PropTypes from 'prop-types'
+import configureStore from '../../../store/configureStore'
+import { loginThunk, logoutThunk } from '../../../redux/session/thunks'
```
Comment (ozalexo) [inline src/screens/Login/Start/StartContainer.js]: logoutThunk only for testing purposes?

**Review by ozalexo — CHANGES_REQUESTED** [review]

**Review by ozalexo — APPROVED** [review]

**Pull request merged**